docker/deploy.sh

```diff
@@ -1,56 +1,150 @@
-    cd docker/swarm
-    ./generate-secret.sh
-
-    set -e # if the secret already exists, don't fail
-
-    mv ./scimsession ../../
-
-    read -p 'Please enter your domain name: ' domain_name
-
-    sed -i.bak s/{YOUR-DOMAIN-HERE}/$domain_name/g docker-compose.yml
-
-    docker stack deploy -c docker-compose.yml op-scim
-    mv docker-compose.yml.bak docker-compose.yml
-
-    docker service logs --raw -f op-scim_scim
+# function used to set up through Docker Swarm
+run_docker_swarm() {
+    echo " "
+    echo "Deploying using Docker Swarm..."
+    echo "(Ctrl+C to cancel)"
+    sleep 3
+
+    # puts the scimsession secret into the Swarm
+    if ! cat $scimsession_file | docker secret create scimsession -
+    then
+        echo " "
+        echo "Failed to create Docker Swarm secret, please investigate the error"
+        sleep 1
+        exit 1
+    fi
+
+    if ! docker stack deploy -c $docker_deploy_file op-scim
+    then
+        echo " "
+        echo "Failed to deploy to Docker Swarm, please investigate the error"
+        sleep 1
+        exit 1
+    fi
+
+    read -p "Do you want to view the logs? [Y/n]: " view_logs
+    if [[ "$view_logs" =~ ^([yY][eE][sS]|[yY])$ ]]
+    then
+        echo " "
+        echo "Press Ctrl+C to quit out of the log view."
+        sleep 2
+        docker service logs --raw -f op-scim_scim 2>/dev/null
+    else
+        echo "Skipping logs..."
+        echo "You can view the logs manually by running: docker service logs --raw -f op-scim_scim"
+    fi
 }
 
-echo 'Please ensure you have read the README before continuing.'
-echo 'Which docker deployment method are you using?'
-read -p '[compose] or [swarm]: ' docker_path
-
-if [ "$docker_path" == "compose" ];
-then 
+# Begin main script
+
+docker_path=$(dirname $(realpath $0))
+
+echo "Initiating 1Password SCIM Bridge Deployment to Docker Swarm"
+echo " "
+echo "Please specify the following options."
+
+while :
+do
+    read -p "Docker Swarm or Docker Compose? [swarm/compose]: " docker_type
+    if [[ "$docker_type" =~ ^(swarm|compose)$ ]]; then
+        break
+    fi
+    echo "$docker_type is not a valid input. Please select either 'swarm' or 'compose'."
+done
+
+
+while :
+do
+    read -p "Fully-qualified domain name (FQDN) you are deploying to [e.g: 'op-scim.example.com']: " domain_name
+    if [[ $domain_name = *.* ]]; then
+        break
+    fi
+    echo "Please enter a fully-qualified domain name."
+done
+
+while :
+do
+    read -p "Path to your scimsession file: " scimsession_file
+    if [[ -f "$scimsession_file" ]]
+    then
+        break
+    fi
+    echo "File '$scimsession_file' does not exist at that path, please try again." >&2
+done
+
+echo " "
+echo "Using the following parameters to deploy the SCIM Bridge"
+echo "Deployment type:" $docker_type
+echo "scimsession file path:" $scimsession_file
+echo "Domain name:" $domain_name
+
+while ! [[ "$proceed" =~ ^([yY][eE][sS]|[yY])$ || "$proceed" =~ ^([nN][oO][nN]) ]]; do
```

Comment:
As the loop exit clause only gets evaluated at the end of the loop, and `no` is checked for within the loop, isn't the `no` on the exit clause redundant?

Reply:
Fixed.